src/components/QuranReader/index.tsx

```diff
@@ -115,10 +119,40 @@ const QuranReader = ({
     }
   };
 
+  /**
+   * select all DOM node that contains data attribute `data-word-location`
```

Comment:
nit: nodes, these nodes are 

1) 
2) 
3) 


then get the uthmani_text -> extract the 

 We need to use `uthmani_text` instead of the rendered text to support unicode based fonts like v1 and v2.